provider/provider_config.go

```diff
@@ -3,38 +3,59 @@ package provider
 import (
 	"github.com/hashicorp/go-cty/cty"
 	"github.com/hashicorp/terraform-provider-kubernetes-alpha/tfplugin5"
+	"github.com/pkg/errors"
 )
 
-// BlockMap is  a the basic building block of a configuration or resource object.
-type BlockMap map[string]cty.Type
-
-// GetConfigObjectType returns the type scaffolding for the provider config object.
-func GetConfigObjectType() cty.Type {
-	return cty.Object(BlockMap{
-		"config_file":          cty.String,
-		"server_side_planning": cty.Bool,
+// getConfigObjectType returns the type scaffolding for the provider config object.
+func getConfigObjectType() cty.Type {
+	return cty.Object(map[string]cty.Type{
+		"host":                   cty.String,
+		"username":               cty.String,
+		"password":               cty.String,
+		"client_certificate":     cty.String,
+		"client_key":             cty.String,
+		"cluster_ca_certificate": cty.String,
+		"config_path":            cty.String,
+		"config_context":         cty.String,
+		"config_context_user":    cty.String,
+		"config_context_cluster": cty.String,
+		"token":                  cty.String,
+		"insecure":               cty.Bool,
+		"server_side_planning":   cty.Bool,
+		"exec": cty.Object(map[string]cty.Type{
+			"api_version": cty.String,
+			"command":     cty.String,
+			"env":         cty.Map(cty.String),
+			"args":        cty.List(cty.String),
+		}),
 	})
 }
 
 // GetProviderConfigSchema contains the definitions of all configuration attributes
 func GetProviderConfigSchema() *tfplugin5.Schema {
-	cfgFileType, _ := cty.String.MarshalJSON()
-	boolType, _ := cty.Bool.MarshalJSON()
+	b, _ := ctyObjectToTfpluginSchema(getConfigObjectType())
```

Comment:
Should we be handling this error here? It seems unlikely that the cty type is going to fail to marshal to JSON so maybe we should just have `ctyObjectToTfpluginSchema` only return the Schema_Block and explicitly silence the error there.

Reply:
Good catch. This is a forgotten quick-hack. I'll fix it.

Reply:
I added error propagation all the way to the plugin API. It should stop Terraform and surface the error if it ever happens.